### PR TITLE
Migrate resources

### PR DESCRIPTION
**Pull request by rubensworks**
This PR does two things:
* Introduces a tool to migrate for filesystems from LegacyResourceMapper to ResourceMapper (#949)
* Uses the tool to migrate our templates to the new system (#948)

### CODE REVIEW

File: bin/lib/migrateLegacyResources.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+function getNewFileName (fullFilePath, suffix) {
+  return fullFilePath + suffix
+}
+
+function shouldMigrateFile (filename) {
+  return filename.indexOf('.') < 0
+}
+
+function shouldMigrateFolder (foldername) {
+  return foldername.indexOf('.') !== 0
```
Comment (RubenVerborgh) [inline bin/lib/migrateLegacyResources.js]: Nit: `foldername[0] !== '.'`

File: bin/lib/migrateLegacyResources.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        console.error(err)
+      }
+    })
+}
+
+async function migrate (path, suffix, verbose) {
+  const files = await promisify(fs.readdir)(path)
+  for (const file of files) {
+    const fullFilePath = Path.join(path, file)
+    const stat = await promisify(fs.lstat)(fullFilePath)
```
Comment (RubenVerborgh) [inline bin/lib/migrateLegacyResources.js]: You probably want to do your promisifications once and save the result.

File: bin/lib/migrateLegacyResources.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ * By default, all extensionless files (that used to be interpreted as Turtle) will now receive a '$.ttl' suffix. */
+/* https://www.w3.org/DesignIssues/HTTPFilenameMapping.html */
+
+module.exports = function (program) {
+  program
+    .command('migrate-legacy-resources')
+    .option('-p, --path <path>', 'Path to the data folder, defaults to \'data/\'')
+    .option('-s, --suffix <path>', 'The suffix to add, defaults to \'$.ttl\'')
+    .option('-v, --verbose', 'Path to the data folder')
+    .description('Migrate extensionless data files pre-5.0.0 to turtle-based data files post-5.0.0')
```
Comment (RubenVerborgh) [inline bin/lib/migrateLegacyResources.js]: "Migrate the data folder from node-solid-server 4 to node-solid-server 5" or similar>

**Comment by rubensworks** [comment]
Processed comments, ready for merge. (But should probably only be merged _after_ #960)

**Pull request merged**